### PR TITLE
feat(components): expose search term on combobox action

### PR DESCRIPTION
**Pull request by darryltec**
## Motivations

There are instances where the action would need to
- Only show up when there's a search term
- Base the label via the search term
- Know the search term on click

This PR enables all of that.

### Added

- Added a `visible` prop
- Exposes the search term on actions

| Empty search term | With search term |
|--|--|
| image | image |

### Changed

- 

### Deprecated

- 

### Removed

- 

### Fixed

- 

### Security

- 

## Testing

---

[In Atlantis we use Github's built in pull request reviews](https://help.github.com/en/articles/about-pull-request-reviews).

Random photo of Atlantis

### CODE REVIEW

File: packages/components/src/Combobox/components/ComboboxAction/ComboboxAction.tsx [hunk 1]
```diff
@@ -1,24 +1,39 @@
...
         className={styles.actionButton}
-        onClick={props.onClick}
+        onClick={e => props.onClick(e, options)}
         type="button"
-        aria-label={props.label}
       >
         <Typography
           element="span"
           size="base"
           textColor="green"
```
Comment (chris-at-jobber) [inline packages/components/src/Combobox/components/ComboboxAction/ComboboxAction.tsx:33]:
```suggestion
          textColor="interactive"
```

😉

Comment (taylorvnoj) [reply packages/components/src/Combobox/components/ComboboxAction/ComboboxAction.tsx]: It actually has to be `green`? `interactive` is on mobile Typography, not web @chris-at-jobber

Comment (chris-at-jobber) [reply packages/components/src/Combobox/components/ComboboxAction/ComboboxAction.tsx:33]: 🤯 OK separate task... "Allow Typography to consume semantic colors" 😄

File: packages/components/src/Combobox/components/ComboboxAction/ComboboxAction.tsx [hunk 2]
```diff
@@ -1,24 +1,39 @@
...
 
-export function ComboboxAction(props: ComboboxActionProps): JSX.Element {
   return (
     <div className={styles.actionContainer}>
       <button
         className={styles.actionButton}
-        onClick={props.onClick}
+        onClick={e => props.onClick(e, options)}
         type="button"
-        aria-label={props.label}
```
Comment (taylorvnoj) [inline packages/components/src/Combobox/components/ComboboxAction/ComboboxAction.tsx:13]: No more `aria-label`?

What used to show up - under ARIA Attributes in the inspector:
Screenshot 2023-12-04 at 5 03 21 PM

What we have now:
Screenshot 2023-12-04 at 5 03 43 PM

**Review by taylorvnoj — COMMENTED** [review]
Ok I _love_ this!  Just a pondering about tests and a question about the aria-label

File: packages/components/src/Combobox/components/ComboboxAction/ComboboxAction.tsx (hunk 2 above)
Comment (darryltec) [reply packages/components/src/Combobox/components/ComboboxAction/ComboboxAction.tsx:13]: Aria's are needed for when it's different from the label (or lack of it). it looks like they're always the same so aria's aren't really needed. Forgot to add a comment on it haha

**Pull request merged**